### PR TITLE
FISH-7647 : Port Fixes for CVE-2017-12617

### PR DESCRIPTION
**Pull request by luiseufrasio**
## Description
Port Fixes for CVE-2017-12617

### Blockers
None

### New tests
None

### Testing Environment
Zulu JDK 11.0.11 on Windows 11 with Maven 3.8.4

## Documentation
None

## Notes for Reviewers
It was avoided to use normalize function in basePath and canonicaPath because it was causing NullPointerException in DAS loading at runtime.

### CODE REVIEW

**Comment by luiseufrasio** [comment]
jenkins test please

**Comment by luiseufrasio** [comment]
jenkins test please

**Review by Pandrex247 — CHANGES_REQUESTED** [review]

File: appserver/web/web-naming/src/main/java/org/apache/naming/resources/FileDirContext.java [hunk 1]
```diff
@@ -209,15 +209,21 @@ public void setDocBase(String docBase) {
         }
 
         // Validate that the document base is an existing directory
-        if (!base.exists() || !base.isDirectory() || !base.canRead())
+        if (!base.exists() || !base.isDirectory() || !base.canRead()) {
             throw new IllegalArgumentException
                     (MessageFormat.format(rb.getString(LogFacade.FILE_RESOURCES_BASE), docBase));
+        }
+
         this.absoluteBase = base.getAbsolutePath();
```
[the remark this answers is not in the document]
Comment (luiseufrasio) [reply appserver/web/web-naming/src/main/java/org/apache/naming/resources/FileDirContext.java]: done

**Comment by luiseufrasio** [comment]
jenkins test please

**Review by Pandrex247 — APPROVED** [review]

**Pull request merged**